### PR TITLE
LG debugging/logcat fixed

### PR DESCRIPTION
**Pull request by duzenko**
## Description

Fixes broken logcat and integrated debugging on LG devices (e.g. LG-H873)

## Related Issues

https://github.com/flutter/flutter/issues/51853

## Tests

I added the following tests:

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, no existing tests failed, so this is *not* a breaking change.

`flutter analyze --flutter-repo` fails with `372351 issues found. (ran in 12.6s; 5796 public members lack documentation)` but I think it's expected with flutter.

As for the tests, I can't think of anything unless you're going to run a test on an actual LG device. But if you had one, it would not have taken 3 months to get to this point and supposedly you would not have broken it in the first place.

### CODE REVIEW

**Comment by duzenko** [comment]
@Hixie What is the problem with authoring? I branched `master`, merged a commit from another branch and for some reason it's all messed up with 11 commits instead of 1. Is it illegal to merge from stable?

**Comment by christopherfujino** [comment]
@duzenko merging stable into master will create conflicts from the cherry picks on stable. If you want a particular stable commit on a branch off of master, you should cherry pick it instead, or manually apply it.

**Comment by duzenko** [comment]
@christopherfujino @Hixie I can't make anything of the failed test logs. What is the problem here?

**Comment by christopherfujino** [comment]
@duzenko it looks like you have unit tests failing that have expectations on which commands are passed to the fake process manager:

```
08:54 +1654 ~8 -2: test/general.shard/android/adb_log_reader_test.dart: AdbLogReader calls adb logcat with expected flags apiVersion 21 [E]                                                            
  Expected: ['adb', '-s', '1234', 'logcat', '-v', 'time', '-T', '11-27 15:39:04.506']
    Actual: [
              'adb',
              '-s',
              '1234',
              'shell',
              '-x',
              'logcat',
              '-v',
              'time',
              '-T',
              '\'11-27 15:39:04.506\''
            ]
     Which: was 'shell' instead of 'logcat' at location [3]
  package:test_api                                              expect
  test/src/fake_process_manager.dart 91:5                       FakeCommand._matches
  test/src/fake_process_manager.dart 305:21                     _SequenceProcessManager.findCommand
  test/src/fake_process_manager.dart 199:37                     FakeProcessManager._runCommand
  test/src/fake_process_manager.dart 220:15                     FakeProcessManager.start
  package:flutter_tools/src/android/android_device.dart 987:50  AdbLogReader.createLogReader
```

**Comment by duzenko** [comment]
> 
> 
> @duzenko it looks like you have unit tests failing that have expectations on which commands are passed to the fake process manager:
> 
> ```
> 08:54 +1654 ~8 -2: test/general.shard/android/adb_log_reader_test.dart: AdbLogReader calls adb logcat with expected flags apiVersion 21 [E]                                                            
>   Expected: ['adb', '-s', '1234', 'logcat', '-v', 'time', '-T', '11-27 15:39:04.506']
>     Actual: [
>               'adb',
>               '-s',
>               '1234',
>               'shell',
>               '-x',
>               'logcat',
>               '-v',
>               'time',
>               '-T',
>               '\'11-27 15:39:04.506\''
>             ]
>      Which: was 'shell' instead of 'logcat' at location [3]
>   package:test_api                                              expect
>   test/src/fake_process_manager.dart 91:5                       FakeCommand._matches
>   test/src/fake_process_manager.dart 305:21                     _SequenceProcessManager.findCommand
>   test/src/fake_process_manager.dart 199:37                     FakeProcessManager._runCommand
>   test/src/fake_process_manager.dart 220:15                     FakeProcessManager.start
>   package:flutter_tools/src/android/android_device.dart 987:50  AdbLogReader.createLogReader
> ```
> 
> I would recommend [running the tools unit tests locally](https://github.com/flutter/flutter/wiki/The-flutter-tool#making-changes-to-the-flutter-tool), and updating one by one the tests that expect the `adb` call without `shell` (there may be other failures too).

How do I rebuild the test? I modified `flutter\packages\flutter_tools\test\general.shard\android\android_device_test.dart` but the test keeps failing as using the previous file version

**Comment by duzenko** [comment]
@Hixie @christopherfujino FYI more than a dozen flutter_tools tests fails on Windows

**Comment by duzenko** [comment]
@christopherfujino @Hixie Will it it take a half year to merge like my other PR's before?

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -975,9 +979,9 @@ class AdbLogReader extends DeviceLogReader {
         'flutter',
       ] else if (apiVersion != null && apiVersion >= kLollipopVersionCode) ...<String>[
         // Otherwise, filter for logs appearing past the present.
-        // Empty `-T` means the timestamp of the logcat command invocation.
+        // '-T 0` means the timestamp of the logcat command invocation.
         '-T',
-        device.lastLogcatTimestamp ?? '',
+        if (device.lastLogcatTimestamp != null) '\'${device.lastLogcatTimestamp}\'' else '0',
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/android/android_device.dart:984]: Was this change necessary for the fix?

**Review by christopherfujino — COMMENTED** [review]
A question

File: packages/flutter_tools/lib/src/android/android_device.dart (hunk 1 above)
Comment (duzenko) [reply packages/flutter_tools/lib/src/android/android_device.dart:984]: Per adb specs, the `-T` has to be followed with either plain number or quoted timestamp.

**Review by christopherfujino — APPROVED** [review]
LGTM. Thanks for this fix @duzenko
